DetectorDescription/DDCMS/plugins/DDHGCalCell.cc

```diff
@@ -0,0 +1,286 @@
+  std::vector<std::string> truncCN    = args.value<std::vector<std::string> >("TruncatedCell");       // Names of truncated cells
+  std::vector<std::string> truncSensN = args.value<std::vector<std::string> >("TruncatedSensitive");  // Names of the sensitive parts for truncated cells
+  std::vector<std::string> extenCN    = args.value<std::vector<std::string> >("ExtendedCell");        // Names of extended cells
+  std::vector<std::string> extenSensN = args.value<std::vector<std::string> >("ExtendedSensitive");   // Names of the sensitive parts for extended cells
+  std::vector<std::string> cornrCN    = args.value<std::vector<std::string> >("CornerCell");          // Names of the corner cells
+  std::vector<std::string> cornrSensN = args.value<std::vector<std::string> >("CornerSensitive");     // Names of the sensitive parts for corner cells
+
+  if ((truncCN.size() != truncSensN.size()) ||
+      (extenCN.size() != extenSensN.size()) ||
+      (cornrCN.size() != cornrSensN.size()))
+    edm::LogWarning("HGCalGeom") << "The number of cells & sensitive differ:"
+				 << " Truncated " << truncCN.size() << ":"
+				 << truncSensN.size() << " Extended "
+				 << extenCN.size() <<":" << extenSensN.size()
+				 << " Corners " << cornrCN.size() << ":"
+				 << cornrSensN.size();
+  if ((truncCN.size() != 3) || (extenCN.size() != 3) ||
+      (cornrCN.size() != 6))
+    edm::LogWarning("HGCalGeom") << "DDHGCalCell: The number of cells does not"
+				 << " match with Standard: Truncated " 
+				 << truncCN.size() << ":3 Extended " 
+				 << extenCN.size() <<":3" << " Corners " 
+				 << cornrCN.size() << ":6";
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalCell: Wafer r " << waferSize
+				<< " T " << waferT << " Cell T " << cellT
+				<< " Cells/Wafer " << nCells << " Material "
+				<< material << "Sensitive Position "
+				<< posSens << " Full Cell: " << fullCN << ":"
+				<< fullSensN;
+  for (unsigned int k=0; k<truncCN.size(); ++k)
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalCell: Truncated Cell[" << k
+				  << "] " << truncCN[k] << ":"
+				  << truncSensN[k];
+  for (unsigned int k=0; k<extenCN.size(); ++k)
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalCell: Extended Cell[" << k
+				  << "] " << extenCN[k] << ":"
+				  << extenSensN[k];
+  for (unsigned int k=0; k<cornrCN.size(); ++k)
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalCell: Corner Cell[" << k
+				  << "] " << cornrCN[k] << ":"
+				  << cornrSensN[k];
+#endif
+
+  dd4hep::Material matter = ns.material(material);
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalCell: " << material 
+				<< " initialized at " << &matter;
+#endif
+  dd4hep::Volume glog1, glog2;
+
+  static const double sqrt3 = std::sqrt(3.0);
+  double R   = waferSize/(3.0*nCells);
+  double r   = 0.5*R*sqrt3;
+  double dx1 = R;
+  double dx2 = 0.5*dx1;
+  double dx3 = 2.5*dx2;
+  double dx4 = 0.5*dx2;
+  double dy1 = r;
+  double dy2 = 0.5*dy1;
+  double dy3 = 1.5*dy1;
```

Comment:
The 3.0 in line 76, 2.5 in line 80, and 1.5 here sure look like arbitrary magic numbers. The code would probably be more maintainable if these were constants with explanatory names so, in the future if the HGCal design changes, they can be more easily found and revised.

Reply:
I do not think this will change in future. I gave the ones which matters